openedx/core/djangoapps/content/search/tests/test_handlers.py

```diff
@@ -0,0 +1,174 @@
+                "type": "course_block",
+                "usage_key": "block-v1:orgA+test_course+test_run+type@sequential+block@test_sequential",
+                "block_id": "test_sequential",
+                "display_name": "sequential",
+                "block_type": "sequential",
+                "context_key": "course-v1:orgA+test_course+test_run",
+                "org": "orgA",
+                "breadcrumbs": [{"display_name": "Test Course"}], "content": {}
+            }
+        ])
+        vertical = self.store.create_child(self.user_id, sequential.location, "vertical", "test_vertical")
+        meilisearch_client.return_value.index.return_value.update_documents.assert_called_with([
+            {
+                "id": "block-v1orgatest_coursetest_runtypeverticalblocktest_vertical-011f143b",
+                "type": "course_block",
+                "usage_key": "block-v1:orgA+test_course+test_run+type@vertical+block@test_vertical",
+                "block_id": "test_vertical",
+                "display_name": "vertical",
+                "block_type": "vertical",
+                "context_key": "course-v1:orgA+test_course+test_run",
+                "org": "orgA",
+                "breadcrumbs": [{"display_name": "Test Course"}, {"display_name": "sequential"}],
+                "content": {}
+            }
+        ])
+
+        # Update the XBlock
+        sequential = self.store.get_item(sequential.location, self.user_id)  # Refresh the XBlock
+        sequential.display_name = "Updated Sequential"
+        self.store.update_item(sequential, self.user_id)
+
+        meilisearch_client.return_value.index.return_value.update_documents.assert_called_with([
+            {
+                "id": "block-v1orgatest_coursetest_runtypesequentialblocktest_sequential-0cdb9395",
+                "type": "course_block",
+                "usage_key": "block-v1:orgA+test_course+test_run+type@sequential+block@test_sequential",
+                "block_id": "test_sequential",
+                "display_name": "Updated Sequential",
+                "block_type": "sequential",
+                "context_key": "course-v1:orgA+test_course+test_run",
+                "org": "orgA",
+                "breadcrumbs": [{"display_name": "Test Course"}], "content": {}
+            },
+            {
+                "id": "block-v1orgatest_coursetest_runtypeverticalblocktest_vertical-011f143b",
+                "type": "course_block",
+                "usage_key": "block-v1:orgA+test_course+test_run+type@vertical+block@test_vertical",
+                "block_id": "test_vertical",
+                "display_name": "vertical",
+                "block_type": "vertical",
+                "context_key": "course-v1:orgA+test_course+test_run",
+                "org": "orgA",
+                "breadcrumbs": [{"display_name": "Test Course"}, {"display_name": "Updated Sequential"}],
+                "content": {}
+            }
+        ])
```

Comment:
@yusuf-musleh, I liked how you used `copy` in your PR. I will update this code with it after the review.